crates/common_utils/src/pii.rs

```diff
@@ -25,6 +25,125 @@ pub const REDACTED: &str = "Redacted";
 /// Type alias for serde_json value which has Secret Information
 pub type SecretSerdeValue = Secret<serde_json::Value>;
 
+/// Strategy for masking a PhoneNumber
+#[derive(Debug)]
+pub struct PhoneNumberStrategy;
+
+/// Phone Number
+#[derive(Debug, serde::Deserialize, serde::Serialize)]
+#[serde(try_from = "String")]
+pub struct PhoneNumber(Secret<String, PhoneNumberStrategy>);
```

Comment:
Do you see a use case for custom formatting?

Currently I'd prefer this format due to its less complexity

e.g
```
struct Student {
    id: u8,
    phone_number: PhoneNumber,
    email_address: Email,
}
```

vs

```
struct Student {
    id: u8,
    phone_number: Secret<PhoneNumber, PhoneNumberStrategy>,
    email_address: Secret<Email, EmailStrategy>,
}
```

Are there any other rules/guidelines that we would be breaking in a contextual sense?

For e.g in this case we are equating that a phone number is always gonna be a secret, should the type phonenumber worry about the formatting of its value or should it only care about its data validity...

For now since we haven't reached any problems that need to separate this concerns, let's keep it this (Since it's easy to extend as well)

Reply:
Fair enough, resolving this conversation.